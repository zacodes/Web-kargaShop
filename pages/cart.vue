<template>
  <div class="container my-2" v-if="hasItem">
    <div class="row">
      <div class="col-9">
        <div class="row">
          <div class="col-12 border text-right text-ml font-weight-normal py-3">
            <div class="row">
              <div class="col-10">
                SEPETİNİZE <b>115,10 TL</b> 'LİK DAHA ÜRÜN EKLEYİN<br>
                <b>KARGO ÜCRETİ ÖDEMEYİN!</b>
              </div>
              <div class="col-2">
                <i class="fa fa-shipping-fast text-xl"></i>
              </div>
            </div>
          </div>
          <div class="col-12 mt-3 border py-3">
            <h2 class="text-lg text-uppercase text-center mb-0">Sepet Detayı</h2>
            <hr class="head-border">
            <table width="100%">
              <tbody>
              <tr>
                <td>
                  <img :src="basketItem.images[0]" width="100">
                </td>
                <td>
                  {{ basketItem.name }}
                </td>
                <td width="200">
                  <b-input-group>
                    <b-input-group-prepend>
                      <b-btn variant="outline-info" @click="itemCount--">-</b-btn>
                    </b-input-group-prepend>
                    <b-form-input type="number" min="0" v-model="itemCount" class="text-center"></b-form-input>
                    <b-input-group-append>
                      <b-btn variant="outline-secondary" @click="itemCount++">+</b-btn>
                    </b-input-group-append>
                  </b-input-group>
                </td>
                <td>
                  {{ basketItem.price * itemCount }} TL + KDV
                </td>
              </tr>
              </tbody>
            </table>
            <a href="/" class="btn btn-light my-4">
              <i class="fa fa-arrow-left"></i> Alışverişe Devam Et
            </a>
          </div>
        </div>
      </div>
      <div class="col-3">
        <div class="row">
          <div class="col-12 border ml-3 pt-3">
            <h2 class="text-ml text-uppercase text-center mb-0">Sepet Özeti</h2>
            <hr class="head-border">
            <div class="row">
              <div class="col-12">
                <div class="row">
                  <div class="col-6">Ara Toplam</div>
                  <div class="col-6 text-right font-weight-bold">
                    {{ basketItem.price * itemCount }} TL
                  </div>
                </div>
              </div>
              <div class="col-12">
                <div class="row">
                  <div class="col-6">KDV</div>
                  <div class="col-6 text-right font-weight-bold">
                    0.00 TL
                  </div>
                </div>
              </div>
              <div class="col-12">
                <div class="row">
                  <div class="col-6">KDV Dahil</div>
                  <div class="col-6 text-right font-weight-bold">
                    {{ basketItem.price * itemCount }} TL
                  </div>
                </div>
              </div>
              <div class="col-12 border my-3"></div>
              <div class="col-12">
                <div class="row">
                  <div class="col-6">Toplam</div>
                  <div class="col-6 text-right font-weight-bold">
                    {{ basketItem.price * itemCount }} TL
                  </div>
                </div>
              </div>
              <div class="col-12 my-3">
                <a href="/uyeliksizLogin" class="btn btn-block btn-warning text-white" style="border-radius: 30px">
                  Alışverişi Tamamla
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="container text-center" v-else>
    eşya yok
  </div>

</template>

<script>
export default {
  data () {
    return {
      hasItem: false,
      itemCount: 0,
      basketItem: null
    }
  },
  mounted () {
    if (localStorage.getItem('itemCount')) {
      this.hasItem = true
      this.itemCount = localStorage.getItem('itemCount')
      this.basketItem = JSON.parse(localStorage.getItem('item'))
    }
  }
}
</script>
