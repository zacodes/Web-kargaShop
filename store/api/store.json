{
    " products " : [
                    {
                            "id" :" 1 " ,
                            "title" : "KANAGAWA - UNISEX TSHIRT" , 
                            "price" : " 34.9 " ,
                            "code" : "K1867" ,
                            "sizes" : " ['S', 'M', 'L', 'XL']  ",
                            "images" : " nuxt/assets/image/product1.jpg " ,
                            "category" : "Unisex T-Shirt"
                    } 
    ],
                

    " categories ": [
                    { 
                        "id": 1, "categoryName": "GİYİM", "categorySubItem": [
                            
                                { "id": "1"  , "categorySubItemName": "Unisex T-shirt" },
                                { "id": "2"  , "categorySubItemName": "Oversize T-Shirt" },
                                { "id": "3"  , "categorySubItemName": "Kapüşonlu T-Shirt" },
                                { "id": "4"  , "categorySubItemName": "Kapüşonsuz Sweat" },
                                { "id": "5"  , "categorySubItemName": "Atkı" },
                                { "id": "6"  , "categorySubItemName": "Buff" },
                                { "id": "7"  , "categorySubItemName": "Sırt Çantası" },
                                { "id": "8"  , "categorySubItemName": "Bere" },
                                { "id": "9"  , "categorySubItemName": "Budget Şapka" },
                                { "id": "10" , "categorySubItemName": "Çorap (Unisex)" }
                        ]
                    },
                    {  "id": "2" , "categoryName": "AKSESUAR" },
                    {  "id": "3" , "categoryName": "HEDİYELİK" },
                    {  "id": "4" , "categoryName": "KUPA VE TERMOS" },
                    {  "id": "5" , "categoryName": "Kadın Cüzdan" }
    ],

    "users":[
        {
            "id":1,
            "name":"ayse@gmail.com", 
            "password":"123456789"
        },
        {
            "id":2,
            "name":"rabia@hotmail.com", 
            "password":"987654321"
        },
        {
            "id":3,
            "name":"zilan@hotmail.com", 
            "password":"123654789"
        }
    ]    

}