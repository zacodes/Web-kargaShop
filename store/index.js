import Vuex from 'vuex'
import Vue from 'vue'



Vue.use(Vuex)


// const createStore = () => {
  // return new Vuex.Store({
    export const state=()=>({  //data
      products: [],
      categories: [],
      cart: [],
cardProduct:[],
      user: '',
      status: '',
      error: '',
})

export const getters={  
  availableProducts(state) {
  return state.products.filter(product => product.stock > 0)
},

getSite: (state) => (id) => {
  return state.products.find(product => product.id === id)



},
aCategory: (state) => (id) => {
  // return state.categories.
  console.log("hello",state.categories)
},
getfilterSite: (state) => (id) => {
  return state.products.filter(product => product.id !== id)
},
cartProducts(state) {
  return state.cart.map(cartItem => {
    const product = state.products.find(product => product.id === cartItem.pid)
    return {
      
      title: product.title,
      price: product.price,
      image: product.image,
      product: product.renk,
      quantity: cartItem.quantity

    }

  })
},

cartTotal(state, getters) {

  return getters.cartProducts.reduce((total, product) => total + product.price * product.quantity, 0)


},
sepet(state, getters) {
  return getters.cartProducts.reduce((urun_sayisi, product) => urun_sayisi + product.quantity, 0)
},

toplam_tutar(state, getters) {
  let toplam = 150
  getters.cartProducts.forEach(product => {
    toplam -= product.price * product.quantity
  });
  if (toplam < 150 & toplam > 0) {
    return "Ücretsiz kargo hakkı kazanmanıza ₺" + toplam + " kaldı!"

    toplam
  }
  else {
    return "Kargo bedava"
  }

},
status(state) {
  return state.status
},
user(state) {
  return state.user
},
error(state) {
  return state.error
    }
}

  export const actions={

      fetchProducts({ commit }) {
        return new Promise((resolve, reject) => {
        
          firebase.database().ref('/products').once('value').then((snapshot) => {
          
            commit('setProducts', snapshot.val())
          });
        })


      },
      fetchCategory({ commit }) {
        return new Promise((resolve, reject) => {
         
          firebase.database().ref('/categories').once('value').then((snapshot) => {
         
            commit('setCategories', snapshot.val())
          });
        })


      },

      addProductToCart(context, product) {
        if (product.stock > 0) {
          const cartItem = context.state.cart.find(item => item.id === product.id)

          if (!cartItem) {
            context.commit('pushProductToCart', product.id)
            // console.log("1")
          }
          else {
            context.commit('incrementItemQuantity', cartItem)
            // console.log("2")

          }
          context.commit('decrementProductStock', product)
          // console.log("3")
          
           var pid=product
          //  var quantity=product.quantity
           var newItem={product}
            var usr=firebase.auth().currentUser
            console.log(usr)
            if (usr){
              firebase.database().ref("/usr"+usr.uid+"cart").push(newItem)
               
            }
           
        }
        
      },

      signupAction({ commit }, payload) {
      
        commit('setStatus', 'loading')
        firebase.auth().createUserWithEmailAndPassword(payload.email, payload.password)
          .then((response) => {
            alert('success')
            commit('setUser', response.user.uid)
            commit('setStatus', 'success')
            commit('setError', null)
          })
          .catch((error) => {
            commit('setStatus', 'failure')
            commit('setError', error.message)
          })
      }
      ,
      signInAction({ commit }, payload) {
        firebase.auth().signInWithEmailAndPassword(payload.email, payload.password).
          then((response) => {
            alert('basarıyla gırıs yapıldı')
            commit('setUser', response.user.uid)
            commit('setStatus', 'success')
            commit('setError', null)
          })
          .catch((error) => {
            commit('setStatus', 'failure')
            commit('setError', error.message)
          })

      },
      signOutAction ( {commit}) {
        firebase.auth().signOut()
          .then((response) => {
            alert('başarıyla çıkıs yaptınız')
            commit('setUser', null)
            commit('setStatus', 'success')
            commit('setError', null)
          })
          .catch((error) => {
            commit('setStatus', 'failure')
            commit('setError', error.message)
          })
      },

      cardAction({commit},payload){
        const productCard={
          title:payload.title,
          imageUrl:payload.image,
          price:payload.price
        }
        firebase.database().ref('productCard').push(productCard).
        then((data)=>{
          console.log(data)
          commit('cardAction',productCard)
        })
        .catch((error)=>{
          console.log(error)
        })

    }}

  export const  mutations= {
      setProducts(state, products) {
        //update products
        state.products = products
      },

      setCategories(state,categories){
          state.categories=categories
      },
      pushProductToCart(state, productId) {
        state.cart.push({
          pid: productId,
          quantity: 1
        })

      },

      incrementItemQuantity(state, cartItem) {
        cartItem.quantity++
      },
      decrementProductStock(state, product) {
        product.stock--
      },
      removeProductFromCart(state, id) {
        let index = state.cart.findIndex(product => product.id);
        console.log(index)
        state.cart.splice(index, 1);
      },

      setUser(state, payload) {
        state.user = payload
      },
      removeUser(state) {
        state.user = null
      },
      setStatus(state, payload) {
        state.status = payload
      },
      setError(state, payload) {
        state.error = payload
      },

      cardMutation(state,payload){
        state.cartProducts=payload
      }


    }


  // })
// }
// export default createStore
   


/*import Vue from "vue"
import Vuex from "vuex"

Vue.use(Vuex);

export const store = new Vuex.Store({
    state:{
        products: [],
    },

    getters:{
        getProducts(state){
            return state.products
        },
    },
}),

export const getters = new Vuex.Store({
    products: (state) => () => {
        return state.data
    },
}),

export const getters = new Vuex.Store({
    setproducts(state, array) {
        state.data = array
    },
}),

*/