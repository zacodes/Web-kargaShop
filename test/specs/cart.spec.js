import { mount, shallowMount } from '@vue/test-utils'
import cart from '../../pages/cart.vue'


describe('cart.vue', () => {

    it('Bileşenimizin class özelliğini kontrol edilmesi1', () => {
        const wrapper = mount(cart)
        expect(wrapper.classes()).toContain('container', 'my-2');
    });

    it('Bileşenimizin class özelliğini kontrol edilmesi2', () => {
        const wrapper = mount(cart)
        expect(wrapper.classes()).toContain('container', 'text-center');
    });

    it('Snapshot özelliği', () => {
        const wrapper = mount(cart)
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('has data', () => {
        const wrapper = mount(cart)
    expect(typeof cart.data).toBe('function')
    });

})


/* 

import { shallowMount, mount, config, RouterLinkStub, createLocalVue } from "@vue/test-utils"
import cart from "../../pages/cart.vue"
import { getters, mutations, actions, state } from "../../store"
import Vuex from "vuex"



describe("cart.vue", () => {

    it('cart.vue Bileşenimizin class özelliğini kontrol edilmesi', () => {
        const wrapper = mount(cart)
        expect(wrapper.classes()).toContain('container');
    });

  describe("Sepet boşken", () => {
    config.mocks["$store"] = {
      state: { ...state },
      getters: {
        ...getters
      },
      mutations: {
        ...mutations
      }
    };
    it("'SEPETİNİZDE ÜRÜN YOK!' yazması", () => {
      let wrapper = shallowMount(cart);
      expect(wrapper.text()).toContain("SEPETİNİZDE ÜRÜN YOK!");
    });
  });

  describe("Sepet Dolu iken", () => {
    let wrapper;
    const localVue = createLocalVue()
    localVue.use(Vuex)

    const store = new Vuex.Store({
      state: {
        inCart: [
          {
            "id": 1,
            "count": 1
          }
        ],

        data: [
          {
            "id": " 1 ",
            "title": "KANAGAWA - UNISEX TSHIRT",
            "price": " 34.9 ",
            "code": "K1867",
            "sizes": " ['S', 'M', 'L', 'XL']  ",
            "images": " nuxt/assets/image/product1.jpg ",
            "category": "Unisex T-Shirt"
          }
        ]
      },

      getters: getters
    })

    wrapper = mount(cart, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      store,
      localVue,
    })

  });
  
    it('Snapshot özelliği', () => {
        const wrapper = mount(cart)
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('has data', () => {
        const wrapper = mount(cart)
    expect(typeof cart.data).toBe('function')
    });
  
    });

    */



    /*
    describe("cart.vue", () => {

      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Sepet Detayı");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Alışverişe Devam Et");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Sepet Özeti");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Ara Toplam");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("KDV Dahil");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Toplam");
      });
  
      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Alışverişi Tamamla");
      });

      it("Render", () => {
        let wrapper = shallowMount(cart);
        expect(wrapper.text()).toContain("Eşya yok");
      });
    });

    */




/*

import { shallowMount, mount, config, RouterLinkStub, createLocalVue } from "@vue/test-utils";
import sepetim from "../../pages/cart.vue";


describe("cart.vue", () => {

  describe("Sepet boşken", () => {
    config.mocks["$store"] = {
      state: { ...state },
      getters: {
        ...getters
      },
      mutations: {
        ...mutations
      }
    };
    it("'SEPETİNİZDE ÜRÜN YOK!' yazması", () => {
      let wrapper = shallowMount(sepetim);
      expect(wrapper.text()).toContain("SEPETİNİZDE ÜRÜN YOK!");
    });
  });

  describe("Sepet Dolu iken", () => {
    let wrapper;
    const localVue = createLocalVue()
    localVue.use(Vuex)

    const store = new Vuex.Store({
      state: {
        inCart: [
          {
          "id": 1,
          "count": 1
          }
        ],

        data: [
          {
          "id" :" 1 " ,
          "title" : "KANAGAWA - UNISEX TSHIRT" , 
          "price" : " 34.9 " ,
          "code" : "K1867" ,
          "sizes" : " ['S', 'M', 'L', 'XL']  ",
          "images" : " nuxt/assets/image/product1.jpg " ,
          "category" : "Unisex T-Shirt"                     
          }
        ]
      },

      getters: getters
    })

    wrapper = mount(sepetim, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      store,
      localVue,
    })
    
    describe("Gözükecekler", () => {
      it("Karga T-shirt", () => {
        expect(wrapper.text()).toContain("Karga T-shirt");
      });
    });

  });
});

*/

