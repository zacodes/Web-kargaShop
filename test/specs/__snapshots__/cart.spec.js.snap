// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart.vue Snapshot özelliği 1`] = `
<div class="container text-center">
  eşya yok
</div>
`;
